github/azure-self-hosted-runners/config.toml.tmpl

```diff
@@ -0,0 +1,121 @@
+[default]
+# This URL is used by instances to send back status messages as they install
+# the github actions runner. Status messages can be seen by querying the
+# runner status in garm.
+# Note: If you're using a reverse proxy in front of your garm installation,
+# this URL needs to point to the address of the reverse proxy. Using TLS is
+# highly encouraged.
+callback_url = "https://${GARM_HOSTNAME}/api/v1/callbacks/status"
+
+# This URL is used by instances to retrieve information they need to set themselves
+# up. Access to this URL is granted using the same JWT token used to send back
+# status updates. Once the instance transitions to "installed" or "failed" state,
+# access to both the status and metadata endpoints is disabled.
+# Note: If you're using a reverse proxy in front of your garm installation,
+# this URL needs to point to the address of the reverse proxy. Using TLS is
+# highly encouraged.
+metadata_url = "https://${GARM_HOSTNAME}/api/v1/metadata"
+
+# This folder is defined here for future use. Right now, we create a SSH
+# public/private key-pair.
+config_dir = "etc"
+
+# Uncomment this line if you'd like to log to a file instead of standard output.
+# log_file = "/tmp/runner-manager.log"
+
+# Enable streaming logs via web sockets. Use garm-cli debug-log.
+enable_log_streamer = false
+
+[metrics]
+# Toggle metrics. If set to false, the API endpoint for metrics collection will
+# be disabled.
+enable = false
+# Toggle to disable authentication (not recommended) on the metrics endpoint.
+# If you do disable authentication, I encourage you to put a reverse proxy in front
+# of garm and limit which systems can access that particular endpoint. Ideally, you
+# would enable some kind of authentication using the reverse proxy, if the built-in auth
+# is not sufficient for your needs.
+disable_auth = false
+
+[jwt_auth]
+# A JWT token secret used to sign tokens.
+# Obviously, this needs to be changed :).
+secret = "${GARM_JWT_SECRET}"
+
+# Time to live for tokens. Both the instances and you will use JWT tokens to
+# authenticate against the API. However, this TTL is applied only to tokens you
+# get when logging into the API. The tokens issued to the instances we manage,
+# have a TTL based on the runner bootstrap timeout set on each pool. The minimum
+# TTL for this token is 24h.
+time_to_live = "8760h"
+
+[apiserver]
+  # Bind the API to this IP
+  bind = "0.0.0.0"
+  # Bind the API to this port
+  port = 9997
+  # Whether or not to set up TLS for the API endpoint. If this is set to true,
+  # you must have a valid apiserver.tls section.
+  use_tls = false
```

Comment:
we have TLS endpoint at the caddy server so this isn't needed?

Reply:
yes, the garm https server wouldn't have a cert for the url